packages/sharing-smart-contract/contracts/modules/Sale.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Apache-2.0
+
+/******************************************************************************
+ * Copyright 2020 IEXEC BLOCKCHAIN TECH                                       *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License");            *
+ * you may not use this file except in compliance with the License.           *
+ * You may obtain a copy of the License at                                    *
+ *                                                                            *
+ *     http://www.apache.org/licenses/LICENSE-2.0                             *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ ******************************************************************************/
+pragma solidity ^0.8.23;
+
+import "../Store.sol";
+
+contract Sale is Store {
+    /***************************************************************************
+     *                        event/modifier                                   *
+     ***************************************************************************/
+    event ProtectedDataAddedForSale(uint256 _collectionId, address _protectedData, uint112 _price);
+    event ProtectedDataRemovedFromSale(uint256 _collectionId, address _protectedData);
+    event ProtectedDataSold(
+        uint256 _collectionIdFrom,
+        uint256 _collectionIdTo,
+        address _protectedData
+    );
+
+    /***************************************************************************
+     *                        Functions                                        *
+     ***************************************************************************/
+    function setProtectedDataForSale(
+        uint256 _collectionId,
+        address _protectedData,
+        uint112 _price
+    ) public onlyProtectedDataInCollection(_collectionId, _protectedData) {
+        protectedDataForSale[_collectionId][_protectedData].onSale = true;
+        protectedDataForSale[_collectionId][_protectedData].price = _price;
+        emit ProtectedDataAddedForSale(_collectionId, _protectedData, _price);
+    }
```

Comment:
we should check
- the data is not available for renting or subscription
- there is no pending rental or subscription on the data